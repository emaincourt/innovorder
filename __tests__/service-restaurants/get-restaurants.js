import Boom from 'boom';
import Joi from 'joi';

beforeEach(() => {
  jest.resetModules();
});

describe('Query validation', () => {
  it('accepts all the query parameters that match the properties of a restaurant', () => {
    const GetRestaurants = require('../../services/service-restaurants/get-restaurants').default;
    expect.assertions(1);
    expect(
      Joi.validate({
        id: 'any-id',
        name: 'any-name',
        preparationDelay: 20,
        rushDelay: 20,
      }, GetRestaurants.config.validate.query).error,
    ).toBeFalsy();
  });
});

describe('Handler', () => {
  it('finds all the restaurants that match the query parameters', async () => {
    expect.assertions(3);
    jest.mock('../../sequelize/models', () => ({
      restaurant: {
        findAll: (query) => {
          expect(query).toEqual({ where: {
            id: 'any-id',
            name: 'any-name',
            preparationDelay: 20,
            rushDelay: 20,
          } });
          return Promise.resolve([{
            id: 'any-id',
            name: 'any-name',
            preparationDelay: 20,
            rushDelay: 20,
          }]);
        },
      },
    }));
    const GetRestaurants = require('../../services/service-restaurants/get-restaurants').default;
    await expect(
      GetRestaurants.handler({
        query: {
          id: 'any-id',
          name: 'any-name',
          preparationDelay: 20,
          rushDelay: 20,
        },
      }, (response) => {
        expect(response).toEqual([{
          id: 'any-id',
          name: 'any-name',
          preparationDelay: 20,
          rushDelay: 20,
        }]);
      }),
    ).resolves.toMatchSnapshot();
  });

  it('forwards the error if one gets thrown during the deletion', async () => {
    expect.assertions(3);
    jest.mock('../../sequelize/models', () => ({
      restaurant: {
        findAll: (query) => {
          expect(query).toEqual({ where: {
            id: 'any-id',
            name: 'any-name',
            preparationDelay: 20,
            rushDelay: 20,
          } });
          return Promise.reject(new Error());
        },
      },
    }));
    const GetRestaurants = require('../../services/service-restaurants/get-restaurants').default;
    await expect(
      GetRestaurants.handler({
        query: {
          id: 'any-id',
          name: 'any-name',
          preparationDelay: 20,
          rushDelay: 20,
        },
      }, (response) => {
        expect(response).toEqual(Boom.wrap(new Error()));
      }),
    ).resolves.toMatchSnapshot();
  });
});
