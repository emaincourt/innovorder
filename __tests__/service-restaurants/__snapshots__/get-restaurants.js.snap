// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Handler finds all the restaurants that match the query parameters 1`] = `undefined`;

exports[`Handler forwards the error if one gets thrown during the deletion 1`] = `undefined`;
